### DISCUSSION TITLE
Cannot generate Ipopt executable

### DISCUSSION THREAD

**Discussion by hideakiv** [coin-or/Ipopt#589] [Q&A] [answered]
I'm using macOS and trying to compile with gnu.
`% ../configure CXX=/usr/local/bin/g++-11 CC=/usr/local/bin/gcc-11 F77=/usr/local/bin/gfortran-11
% make
% sudo make install`

gives me these outputs.
for make:
make[2]: Nothing to be done for 'all'.
make[2]: Nothing to be done for 'all-am'.
make[1]: Nothing to be done for 'all-am'.
for make install:
make[3]: Nothing to be done for 'install-exec-am'.
make[3]: Nothing to be done for 'install-data-am'.
make[2]: Nothing to be done for 'install-exec-am'.

**Comment by svigerske** [comment, marked as answer]
This looks like the build worked fine. If one invokes `make` a second time, it will have nothing to do, so that's what it says.
You may want to start from scratch and provide a full log output.

If one doesn't [provide ASL](https://coin-or.github.io/Ipopt/INSTALL.html#EXTERNALCODE_ASL), then no executable is build.

**Reply by hideakiv** [reply]
I am trying to use Ipopt in Pyomo. Will the executable be built if I provide HSL? How can I check whether HSL is installed?

**Reply by svigerske** [reply]
To use Pyomo, you need to build Ipopt with AMPL interface (ASL). To do that, you should build and install [ThirdParty-ASL](https://github.com/coin-or-tools/ThirdParty-ASL) first: https://coin-or.github.io/Ipopt/INSTALL.html#EXTERNALCODE_ASL

The output of Ipopt's configure will say whether it found ASL.

There is also a helper script that builds Ipopt including dependencies: https://coin-or.github.io/Ipopt/INSTALL.html#COINBREW

HSL is recommended for performance reasons, but it has nothing to do with whether the Ipopt executable is build. The Ipopt executable is only useful when one has AMPL nl files (like Pyomo produces), so it is only build if ASL is available.
